java/token-requestor-client/src/main/java/no/bankaxept/epayment/client/tokenrequestor/TokenRequestorClient.java

```diff
@@ -0,0 +1,59 @@
+package no.bankaxept.epayment.client.tokenrequestor;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import no.bankaxept.epayment.client.base.*;
+
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Flow;
+
+public class TokenRequestorClient {
+    private final BaseClient baseClient;
+
+    private final static String ENROLMENT_URL = "/payment-tokens";
+    private final static String DELETION_URL = "/payment-tokens/%s/deletion"; //Token id in path
+
+    private final ObjectMapper objectMapper = new ObjectMapper()
+            .registerModule(new JavaTimeModule())
+            .disable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS);
+
+    private final Executor executor = Executors.newSingleThreadExecutor();
+
+    public TokenRequestorClient(BaseClient baseClient) {
+        this.baseClient = baseClient;
+    }
+
+    public TokenRequestorClient(String baseurl, String apimKey, String username, String password) {
+        this.baseClient = new BaseClient(baseurl, apimKey, username, password);
+    }
+
+    public Flow.Publisher<RequestResponse<EnrolCardResponse>> enrol(EnrolCardRequest request, String correlationId) {
```

Comment:
There should be no body in the response, so we should fix that first. Then you don't need the `RequestResponse`. We did a similar change on the Merchant API not long ago, but we must have forgotten to do it on this one.

Reply:
Is it never gonna be needed? Removed it now

Reply:
Is the tokenId supplied in a callback now or something?

Reply:
Yes, there should be a callback.

Reply:
`components.schemas.EnrolCardCallbackData` should have a tokenId similar to `components.schemas.PaymentRequestCallbackData` having a paymentId.

Reply:
I created an [issue](https://github.com/techcloud0/bankaxept-epayment-development-kit/issues/17) to update this